torch_geometric/nn/conv/x_conv.py

```diff
@@ -129,12 +129,13 @@ def forward(self, x: Tensor, pos: Tensor, batch: Optional[Tensor] = None):
 
         if self.dilation > 1:
             dil = self.dilation
-            index = torch.randint(K * dil, (N, K), dtype=torch.long,
-                                  device=row.device)
-            arange = torch.arange(N, dtype=torch.long, device=row.device)
-            arange = arange * (K * dil)
-            index = (index + arange.view(-1, 1)).view(-1)
-            row, col = row[index], col[index]
+            randnum = torch.randperm(K * dil, dtype=torch.long,
```

Comment:
It looks like the same indices are selected for all nodes. Can we prevent that, e.g. via:
```python
index = torch.randperm(N * K * dil).view(N, K * dil)[N, :K] % (K * dil)
```